### DISCUSSION TITLE
How to pass a vector<u8> using ts-sdk??

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#433] [Questions] [answered]
i am trying to pass a coin struct name as vector<u8>,
how can i pass it?? 
0x1::aptos::AptosCoin

**Comment by 0xjunha** [comment, marked as answer]
You can use [`serializeStr`](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/bcs/serializer.ts#L99) and [`toUint8Array`](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/bcs/serializer.ts#L249) methods of BCS serializer

- [test cases](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/unit/serializer.test.ts)

**Comment by ajaythxkur** [comment]
@0xjunha can you check this out??
` const coin = result[0];
                    if(coin.token_standard === "v1"){
                        const serializer = new Serializer();
                        serializer.serializeStr(coin.asset_type);
                        coin_type = serializer.toUint8Array();
                    }`
my string :
0xe9142d6a078718688cb6804e3379157f8a19378df1cc97d8de10df5ed54c5869::simpu_coin::SimpuCoin

and this is string being returned from view function
Y0xe9142d6a078718688cb6804e3379157f8a19378df1cc97d8de10df5ed54c5869::simpu_coin::SimpuCoin

and coin types are not matching
ECOIN_TYPE_DO_NOT_MATCH

**Comment by 0xjunha** [comment]
Do you mean there's a `Y` at the beginning of the string? Can you share the full context?

**Reply by ajaythxkur** [reply]
Yes there is Y in the start

**Reply by ajaythxkur** [reply]
I'm passing without y but getting y from view function

**Reply by ajaythxkur** [reply]
#[view]
    public fun coin_as_string_in_token(object_address: address): String acquires CoinInfo {
        assert_coin_info_exists(object_address);
        utf8(borrow_global<CoinInfo>(object_address).coin_type)
    }

**Reply by ajaythxkur** [reply]
Its working fine in tests. There is some issue with the above code while serializing

**Reply by ajaythxkur** [reply]
@0xjunha i deserialized it and its giving the correct value
`const serializer = new Serializer();
                        serializer.serializeStr(coin.asset_type);
                        coin_type = serializer.toUint8Array();
                        const deserializer = new Deserializer(serializer.toUint8Array());
                        console.log(deserializer.deserializeStr())`
                        
                        is it because i am trying to pass vector<u8> to Option<vector<u8>>??
